pkg/common/cns-lib/volume/manager.go

```diff
@@ -720,6 +730,128 @@ func (m *defaultManager) ExpandVolume(ctx context.Context, volumeID string, size
 			log.Errorf("ConnectCns failed with err: %+v", err)
 			return err
 		}
+		if m.idempotencyHandlingEnabled && m.operationStore != nil {
+			return m.expandVolumeWithImprovedIdempotency(ctx, volumeID, size)
+		}
+		return m.expandVolume(ctx, volumeID, size)
+
+	}
+	start := time.Now()
+	err := internalExpandVolume()
+	if err != nil {
+		prometheus.CnsControlOpsHistVec.WithLabelValues(prometheus.PrometheusCnsExpandVolumeOpType,
+			prometheus.PrometheusFailStatus).Observe(time.Since(start).Seconds())
+	} else {
+		prometheus.CnsControlOpsHistVec.WithLabelValues(prometheus.PrometheusCnsExpandVolumeOpType,
+			prometheus.PrometheusPassStatus).Observe(time.Since(start).Seconds())
+	}
+	return err
+}
+
+// createVolumeWithoutIdempotency invokes CNS ExpandVolume.
+func (m *defaultManager) expandVolume(ctx context.Context, volumeID string, size int64) error {
```

Comment:
This function is just refactored code to the old logic. There are no changes here to the existing logic here.